### PR TITLE
MG-104: Rename/Replace filed owner to domain

### PR DESCRIPTION
**Pull request by arvindh123**
# What type of PR is this?
This is a refactor because it changes the following functionality: 
-  Rename/Replace filed owner to domain in all services

## Which issue(s) does this PR fix/relate to?

- Resolves absmach/magistrala#1953 

## Have you included tests for your changes?

Yes, WIP

## Did you document any new/modified feature?

Yes, WIP

Notes: 
Should be merged after
- https://github.com/absmach/magistrala-old/pull/194
- https://github.com/absmach/magistrala-old/pull/200
- https://github.com/absmach/magistrala-old/pull/258
- https://github.com/absmach/magistrala-old/pull/262

### CODE REVIEW

File: api/openapi/things.yml [hunk 1]
```diff
@@ -382,8 +381,7 @@ paths:
         - Channels
       summary: Creates new channel
       description: |
-        Creates new channel. User identified by the provided access token will
-        be the channel's owner.
+        Creates new channel. Domain identified by the provided access token will.
```
Comment (dborovcanin) [inline api/openapi/things.yml:384]: Fix.

File: things/api/http/endpoints_test.go [hunk 2]
```diff
@@ -456,33 +456,6 @@ func TestListThings(t *testing.T) {
...
 			err:    apiutil.ErrValidation,
 		},
-		{
-			desc:  "list things with owner_id",
-			token: validToken,
-			listThingsResponse: mgclients.ClientsPage{
-				Page: mgclients.Page{
-					Total: 1,
-				},
-				Clients: []mgclients.Client{client},
```
Comment (dborovcanin) [inline things/api/http/endpoints_test.go:466]: These tests are still relevant, just replace the owner with the domain.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: things/api/http/endpoints_test.go (hunk 2 above)
Comment (arvindh123) [reply things/api/http/endpoints_test.go:466]: @dborovcanin Yesterday, I missed to push one commit
This is no more needed, It was removed
Things list with owner_id (domain_id) will not work

File: api/openapi/things.yml (hunk 1 above)
Comment (arvindh123) [reply api/openapi/things.yml:384]: @dborovcanin Fixed

Comment (dborovcanin) [reply api/openapi/things.yml:384]: This is not fixed.

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**